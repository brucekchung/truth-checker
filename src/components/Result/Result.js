import React from 'react'
import { connect } from 'react-redux'
import { cleanArticle } from '../../cleaner'

export const Result = ({ rawContent, cleanArticle }) => {

  if(cleanArticle) {
    const { title, type, author, date, siteName, text} = cleanArticle
    return (
      <div>
        <p>title: {title}</p>
        <p>type: {type}</p>
        <p>author: {author}</p>
        <p>date: {date}</p>
        <p>siteName: {siteName}</p>
        <p>text: {text}</p>
      </div>
    )
  } else {
    //return null later
    return (
      <div>
        Result container
        example url: 
        https://www.cnn.com/travel/article/passenger-planes-future-look/index.html
        http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html
      </div>
    )
  }
}

const mapState = (state) => ({
  rawContent: state.rawContent,
  cleanArticle: state.cleanArticle,
})

export default connect(mapState)(Result)