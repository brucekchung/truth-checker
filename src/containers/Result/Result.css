.Result {
  background: white;
  width: 95%;
  margin: 0 auto;
  margin-top: 15px;
  border-radius: 14px;
  padding: 15px 0 15px 0;
}

.original-text {
  height: 300px;
  overflow: auto;
  width: 95%;
  margin: 5px auto;
  border-radius: 14px;
  margin-bottom: 10px;
  border: 1px solid #535252;
}