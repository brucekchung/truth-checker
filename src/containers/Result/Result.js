import React from 'react'
import { connect } from 'react-redux'
import { string, oneOfType, object } from 'prop-types'
import { combinedScore } from '../../helper.js'

export const Result = ({ error, cleanArticle, rating }) => {
  if (error) {
    return (
      <div>
        <p>Result could not be displayed</p>
        <p>error code: {error.errorCode}</p>
        <p>error message: {error.error}</p>
      </div>
    )

  } else if (cleanArticle) {
    const {title, type, author, date, siteName, text} = cleanArticle

    return (
      <div>
        {
          !rating &&
          <h2>Rating: Pending</h2>
        }

        {
          rating &&
          <div className="result-info">
            <h2>Rating: {combinedScore(rating)}</h2>
            <h3>Organization: {rating.website}</h3>
            <h3>Author: search {rating.author.searchLength}</h3>
            <h3>
              Content: 
              anger {rating.article.anger}, 
              fear {rating.article.fear}, 
              analytical {rating.article.analytical}
            </h3>
          </div>
        }

        <div className="article-info">
          <p>title: {title}</p>
          <p>type: {type}</p>
          <p>author: {author}</p>
          <p>date: {date}</p>
          <p>site name: {siteName}</p>
          <p>text: {text}</p>
        </div>
      </div>
    )

  } else {
    //return null or keep page? can fine tune examples
    return (
      <div>
        <p>example url: </p>
        <p>https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400</p>
        <p>https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade</p>
        <p>https://www.cnn.com/travel/article/passenger-planes-future-look/index.html</p>
        <p>http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html</p>
        <p>https://www.washingtonpost.com/?utm_term=.4205470b3748</p>
        <p>https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing</p>
        <p>http://www.bbc.com/news/world-us-canada-43088644</p>
        <p>http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook</p>
        <p>https://www.rd.com/advice/parenting/stop-bullying-strategy/</p>
      </div>
    )
  }
}

Result.propTypes = {
  error: object,
  cleanArticle: object,
  rating: oneOfType([object, string])
}

export const mapState = (state) => ({
  error: state.error,
  cleanArticle: state.cleanArticle,
  rating: state.rating
})

export default connect(mapState)(Result)