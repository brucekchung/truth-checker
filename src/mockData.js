export const mock = {
  article: {
    title: 'Trump4lyfe',
    type: 'article',
    objects: [{
      author: 'Bruce',
      date: 'today',
      siteName: 'Trump4prez4eva.com',
      text: '2016 was a great year',
      pageUrl: 'www.trumplyfe.com'
    }]
  },
  watson: {
    document_tone: {
      tone_categories: [{
        tones: [
          {score: 0.1},
          {score: 0.2},
          {score: 0.3},
          {score: 0.3},
        ]
      },
      {
        tones: [{score: 0.5}]
      }
      ]
    }
  },
  author: {
    count: 100,
    results: [{citedCount: 1}, {citedCount: 3}]
  },
  site: [
    {
      RatingIcons: [{Url: "http://www.bbb.org/ratingicons/lg-A+.png", DeviceType: "web"}]
    }
  ],
  rating: {
    website: 'A+',
    author: {searchLength: 1300},
    article: {neutral: 0.4}
  }
}

export const mockApiData = {
  url: {
    author: 'Bruce',
    text: 'stuff'
  }
}

export const mockState = {
  result: {
    error: 'stuff',
    cleanArticle: 'stuff',
    rating: 'A+'
  },
  search: {
    cleanArticle: 'stuff'
  }
}