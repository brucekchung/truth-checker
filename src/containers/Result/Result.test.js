import React from 'react'
import { Result, mapState } from './Result'
import { shallow } from 'enzyme'
import { mockState } from '../../mockData'

describe('Result', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Result />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should display an error if an error object is passed', () => {
    const error = {
      errorCode: 500,
      error: 'stuff'
    }
    const wrapper = shallow(<Result error={error} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('mapState should return the correct data', () => {
    const actual = mapState(mockState.result)
    const expected = {
      error: 'stuff',
      cleanArticle: 'stuff',
      rating: 'A+'
    }

    expect(actual).toEqual(expected)
  })
})