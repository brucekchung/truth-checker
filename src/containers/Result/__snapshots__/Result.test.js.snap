// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should display an error if an error object is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    error={
        Object {
            "error": "stuff",
            "errorCode": 500,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Result could not be displayed
</p>,
        <p>
          error code: 
          500
</p>,
        <p>
          error message: 
          stuff
</p>,
      ],
      "className": "Result",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Result could not be displayed",
        },
        "ref": null,
        "rendered": "Result could not be displayed",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "error code: ",
            500,
          ],
        },
        "ref": null,
        "rendered": Array [
          "error code: ",
          500,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "error message: ",
            "stuff",
          ],
        },
        "ref": null,
        "rendered": Array [
          "error message: ",
          "stuff",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Result could not be displayed
</p>,
          <p>
            error code: 
            500
</p>,
          <p>
            error message: 
            stuff
</p>,
        ],
        "className": "Result",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Result could not be displayed",
          },
          "ref": null,
          "rendered": "Result could not be displayed",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "error code: ",
              500,
            ],
          },
          "ref": null,
          "rendered": Array [
            "error code: ",
            500,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "error message: ",
              "stuff",
            ],
          },
          "ref": null,
          "rendered": Array [
            "error message: ",
            "stuff",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Result should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          example url: 
</p>,
        <p>
          https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400
</p>,
        <p>
          https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade
</p>,
        <p>
          https://www.cnn.com/travel/article/passenger-planes-future-look/index.html
</p>,
        <p>
          http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html
</p>,
        <p>
          https://www.washingtonpost.com/?utm_term=.4205470b3748
</p>,
        <p>
          https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing
</p>,
        <p>
          http://www.bbc.com/news/world-us-canada-43088644
</p>,
        <p>
          http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook
</p>,
        <p>
          https://www.rd.com/advice/parenting/stop-bullying-strategy/
</p>,
      ],
      "className": "Result",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "example url: ",
        },
        "ref": null,
        "rendered": "example url: ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400",
        },
        "ref": null,
        "rendered": "https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade",
        },
        "ref": null,
        "rendered": "https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.cnn.com/travel/article/passenger-planes-future-look/index.html",
        },
        "ref": null,
        "rendered": "https://www.cnn.com/travel/article/passenger-planes-future-look/index.html",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html",
        },
        "ref": null,
        "rendered": "http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.washingtonpost.com/?utm_term=.4205470b3748",
        },
        "ref": null,
        "rendered": "https://www.washingtonpost.com/?utm_term=.4205470b3748",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing",
        },
        "ref": null,
        "rendered": "https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "http://www.bbc.com/news/world-us-canada-43088644",
        },
        "ref": null,
        "rendered": "http://www.bbc.com/news/world-us-canada-43088644",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook",
        },
        "ref": null,
        "rendered": "http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://www.rd.com/advice/parenting/stop-bullying-strategy/",
        },
        "ref": null,
        "rendered": "https://www.rd.com/advice/parenting/stop-bullying-strategy/",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            example url: 
</p>,
          <p>
            https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400
</p>,
          <p>
            https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade
</p>,
          <p>
            https://www.cnn.com/travel/article/passenger-planes-future-look/index.html
</p>,
          <p>
            http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html
</p>,
          <p>
            https://www.washingtonpost.com/?utm_term=.4205470b3748
</p>,
          <p>
            https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing
</p>,
          <p>
            http://www.bbc.com/news/world-us-canada-43088644
</p>,
          <p>
            http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook
</p>,
          <p>
            https://www.rd.com/advice/parenting/stop-bullying-strategy/
</p>,
        ],
        "className": "Result",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "example url: ",
          },
          "ref": null,
          "rendered": "example url: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400",
          },
          "ref": null,
          "rendered": "https://www.wsj.com/articles/like-peter-thiel-others-feel-alienated-by-silicon-valley-groupthink-1518962400",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade",
          },
          "ref": null,
          "rendered": "https://www.bloomberg.com/news/articles/2018-01-09/trump-loving-farmers-let-him-know-that-they-also-love-free-trade",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.cnn.com/travel/article/passenger-planes-future-look/index.html",
          },
          "ref": null,
          "rendered": "https://www.cnn.com/travel/article/passenger-planes-future-look/index.html",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html",
          },
          "ref": null,
          "rendered": "http://www.independent.co.uk/voices/holocaust-israel-poland-history-difficult-acknowledge-netanyahu-jewish-polish-government-a8212071.html",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.washingtonpost.com/?utm_term=.4205470b3748",
          },
          "ref": null,
          "rendered": "https://www.washingtonpost.com/?utm_term=.4205470b3748",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing",
          },
          "ref": null,
          "rendered": "https://www.theonion.com/nra-calls-for-teachers-to-keep-loaded-gun-pointed-at-cl-1819575763?utm_content=Main&utm_campaign=SF&utm_source=Facebook&utm_medium=SocialMarketing",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "http://www.bbc.com/news/world-us-canada-43088644",
          },
          "ref": null,
          "rendered": "http://www.bbc.com/news/world-us-canada-43088644",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook",
          },
          "ref": null,
          "rendered": "http://www.dancemagazine.com/peter-martins-nycb-investigation-results-2535425336.html?utm_campaign=RebelMouse&socialux=facebook&share_id=3329042&utm_medium=social&utm_content=Dance+Magazine&utm_source=facebook",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://www.rd.com/advice/parenting/stop-bullying-strategy/",
          },
          "ref": null,
          "rendered": "https://www.rd.com/advice/parenting/stop-bullying-strategy/",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
